sdcflows/fieldmaps.py

```diff
@@ -404,16 +404,20 @@ def __attrs_post_init__(self):
         # Register this estimation method
         if not self.bids_id:
             # If not manually set, try to get it from BIDS metadata
-            bids_ids = set(
-                [
-                    f.metadata.get("B0FieldIdentifier")
-                    for f in self.sources
-                    if f.metadata.get("B0FieldIdentifier")
-                ]
-            )
-            if len(bids_ids) > 1:
+            b0_ids = [
+                listify(f.metadata.get("B0FieldIdentifier"))
+                for f in self.sources
+                if f.metadata.get("B0FieldIdentifier")
+            ]
+            bids_ids = set().intersection(*b0_ids)
```

Comment:
Merged too fast, the intersection of an empty set with anything will always be an empty set.

Reply:
Oh, right. Guess we could do:

```suggestion
            all_ids = set().union(*b0_ids)
            bids_ids = all_ids.intersection(*b0_ids)
```

Reply:
I understand, is `a[0].intersection(*a)`  less offensive ?

Reply:
in fact `set.intersection(*a)` works if called on the class and not an empty instance.